src/webgl/primitives.js

```diff
@@ -1203,4 +1203,352 @@ p5.RendererGL.prototype.line = function() {
+    this.immediateMode._curveVertex.push(arguments[1]);
+    if (this.immediateMode._curveVertex.length === 8) {
+      this.isCurve = true;
+      w_x = this._bezierToCatmull([
+        this.immediateMode._curveVertex[0],
+        this.immediateMode._curveVertex[2],
+        this.immediateMode._curveVertex[4],
+        this.immediateMode._curveVertex[6]
+      ]);
+      w_y = this._bezierToCatmull([
+        this.immediateMode._curveVertex[1],
+        this.immediateMode._curveVertex[3],
+        this.immediateMode._curveVertex[5],
+        this.immediateMode._curveVertex[7]
+      ]);
+      for (i = 1; i < LUTLength; i++) {
+        _x =
+          w_x[0] * this._lookUpTableBezier[i][0] +
+          w_x[1] * this._lookUpTableBezier[i][1] +
+          w_x[2] * this._lookUpTableBezier[i][2] +
+          w_x[3] * this._lookUpTableBezier[i][3];
+        _y =
+          w_y[0] * this._lookUpTableBezier[i][0] +
+          w_y[1] * this._lookUpTableBezier[i][1] +
+          w_y[2] * this._lookUpTableBezier[i][2] +
+          w_y[3] * this._lookUpTableBezier[i][3];
+        this.vertex(_x, _y);
+      }
+      this.immediateMode._curveVertex.shift();
+      this.immediateMode._curveVertex.shift();
+    }
+  } else if (arguments.length === 3) {
+    this.immediateMode._curveVertex.push(arguments[0]);
+    this.immediateMode._curveVertex.push(arguments[1]);
+    this.immediateMode._curveVertex.push(arguments[2]);
+    if (this.immediateMode._curveVertex.length === 12) {
+      this.isCurve = true;
+      w_x = this._bezierToCatmull([
+        this.immediateMode._curveVertex[0],
+        this.immediateMode._curveVertex[3],
+        this.immediateMode._curveVertex[6],
+        this.immediateMode._curveVertex[9]
+      ]);
+      w_y = this._bezierToCatmull([
+        this.immediateMode._curveVertex[1],
+        this.immediateMode._curveVertex[4],
+        this.immediateMode._curveVertex[7],
+        this.immediateMode._curveVertex[10]
+      ]);
+      w_z = this._bezierToCatmull([
+        this.immediateMode._curveVertex[2],
+        this.immediateMode._curveVertex[5],
+        this.immediateMode._curveVertex[8],
+        this.immediateMode._curveVertex[11]
+      ]);
+      for (i = 1; i < LUTLength; i++) {
+        _x =
+          w_x[0] * this._lookUpTableBezier[i][0] +
+          w_x[1] * this._lookUpTableBezier[i][1] +
+          w_x[2] * this._lookUpTableBezier[i][2] +
+          w_x[3] * this._lookUpTableBezier[i][3];
+        _y =
+          w_y[0] * this._lookUpTableBezier[i][0] +
+          w_y[1] * this._lookUpTableBezier[i][1] +
+          w_y[2] * this._lookUpTableBezier[i][2] +
+          w_y[3] * this._lookUpTableBezier[i][3];
+        _z =
+          w_z[0] * this._lookUpTableBezier[i][0] +
+          w_z[1] * this._lookUpTableBezier[i][1] +
+          w_z[2] * this._lookUpTableBezier[i][2] +
+          w_z[3] * this._lookUpTableBezier[i][3];
+        this.vertex(_x, _y, _z);
+      }
+      this.immediateMode._curveVertex.shift();
```

Comment:
I understand that such a short loop seems like overkill but this feels a bit odd.